test/test_modules.py

```diff
@@ -483,6 +483,22 @@ def test_pip_package(host):
     assert int(outdated['latest'].split('.')[0]) > 2
 
 
+def test_x509(host):
+    host.check_output('apt-get -y update')
+    host.check_output('apt-get -y install curl')
+    host.check_output(
+        'curl -so /tmp/crt https://letsencrypt.org/certs/isrgrootx1.pem.txt'
+    )
+    assert host.file('/tmp/crt').is_file
+    assert 'ISRG Root' in host.x509('/tmp/crt').subject
+    in_ten_days = datetime.datetime.now() + datetime.timedelta(days=10)
+    assert host.x509('/tmp/crt').enddate > in_ten_days
```

Comment:
Maybe generate a certificate in dockerfile with a fixed end date for this test ?